software/seqtk/subseq/main.nf

```diff
@@ -0,0 +1,45 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process SEQTK_SUBSEQ {
+    tag '$sequences'
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:[:], publish_by_meta:[]) }
+
+    conda (params.enable_conda ? "bioconda::seqtk=1.3" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/seqtk:1.3--h5bf99c6_3"
+    } else {
+        container "quay.io/biocontainers/seqtk:1.3--h5bf99c6_3"
+    }
+
+    input:
+    path sequences
+    path filter_list
+
+    output:
+    path "*.gz",          emit: sequences
+    path "*.version.txt", emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def ext = "fa"
+    if ("$sequences" ==~ /.+\.fq|.+\.fq.gz|.+\.fastq|.+\.fastq.gz/) {
```

Comment:
Nice! This works?! 😆

Reply:
Yup, I checked in the Groovy Playground :) -- https://groovy-playground.appspot.com/